.github/workflows/build-ci.yml

```diff
@@ -16,56 +16,102 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - name-prefix: "C++ without MPI"
+          - name-prefix: "Python 3.6 without MPI"
             enable-mpi: false
-            configure-options: "--without-python"
-            python-version: "none"
-          - name-prefix: "C++ with MPI"
-            enable-mpi: true
-            configure-options: "--without-python --with-mpi"
-            python-version: "none"
-          - name-prefix: "Python 3.6"
-            enable-mpi: false
-            configure-options: "--with-python --enable-maintainer-mode"
+            configure-options: "--without-mpi"
             python-version: "3.6"
-          - name-prefix: "Python 3.9"
+          - name-prefix: "Python 3.9 without MPI"
             enable-mpi: false
-            configure-options: "--with-python --enable-maintainer-mode"
+            configure-options: "--without-mpi"
+            python-version: "3.9"
+          - name-prefix: "Python 3.6 with MPI"
+            enable-mpi: true
+            configure-options: "--with-mpi"
+            python-version: "3.6"
+          - name-prefix: "Python 3.9 with MPI"
+            enable-mpi: true
+            configure-options: "--with-mpi"
             python-version: "3.9"
 
     steps:
-    - name: Install Dependencies
+    - name: Define common environment variables
       run: |
-        sudo apt-get update -y
-        sudo apt-get install -y autoconf automake libaec-dev libctl-dev libfftw3-dev libgdsii-dev libgsl-dev libharminv-dev libhdf5-dev libtool mpb mpb-dev swig
-    - name: Install MPI
+        echo "CPPFLAGS=-I${HOME}/local/include" >> $GITHUB_ENV
+        echo "LDFLAGS=-L${HOME}/local/lib" >> $GITHUB_ENV
+        echo "HDF5_BASE_CPPFLAGS=-I/usr/include/hdf5" >> $GITHUB_ENV
+        echo "HDF5_BASE_LDFLAGS=-L/usr/lib/x86_64-linux-gnu/hdf5" >> $GITHUB_ENV
+        echo "GEN_CTL_IO=${HOME}/local/bin/gen-ctl-io" >> $GITHUB_ENV
+
+    - name: Define more common environment variables
+      run: |
+        echo "HDF5_SERIAL_CPPFLAGS=${HDF5_BASE_CPPFLAGS}/serial" >> $GITHUB_ENV
+        echo "HDF5_PARALLEL_CPPFLAGS=${HDF5_BASE_CPPFLAGS}/openmpi" >> $GITHUB_ENV
+        echo "HDF5_SERIAL_LDFLAGS=${HDF5_BASE_LDFLAGS}/serial" >> $GITHUB_ENV
+        echo "HDF5_PARALLEL_LDFLAGS=${HDF5_BASE_LDFLAGS}/openmpi" >> $GITHUB_ENV
+
+    - name: Install common dependencies
+      run: |
+        sudo apt-get -y update
+        sudo apt-get -y install autoconf automake libfftw3-dev libgsl-dev liblapack-dev guile-3.0-dev libpng-dev libtool swig
+        git clone https://github.com/NanoComp/libctl.git libctl-src
+        (cd libctl-src && git checkout master && sh autogen.sh --prefix=${HOME}/local --enable-shared && make -j 2 && make install)
+        git clone https://github.com/NanoComp/harminv.git
+        (cd harminv && git checkout master && sh autogen.sh --prefix=${HOME}/local --enable-shared && make -j 2 && make install)
+        git clone https://github.com/NanoComp/mpb.git
```

Comment:
Consider the checkout action instead of cloning the full repo:
https://github.com/actions/checkout

Reply:
Could also just do a `git clone --depth 1` (shallow clone).